### PR TITLE
Setup repo

### LINKED ISSUE #1: Setup repo
- [x]  Contributing
- [x]  Init README
- [x]  Add gitignore
- [x]  Setup PR & issue template
- [x]  Branches rules
- [x]  Licence

### PR DESCRIPTION
**Pull request by c1-ra**
## Description
Setup of the server repository

## Changes
/

## Notes
/

### CODE REVIEW

**Comment by moreirathomas** [comment]
Suggestion: we may want to setup the CI with this PR too.
See `.github/workflows/ci.yml` and `.golangci.yml` from the runner for reference.
Including `make` scripts could be nice too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- Avoid _when possible_ lines of 80+ chars
+- Every file should end with an empty line
+
+## Git
+
+- Branch names must follow this pattern: `<type>/name-of-related-issue`
+- Commit messages must be 50 chars max (add a description if necessary)
+- Commit messages must start with a verb in the present simple, such as `Add ...`, `Implement ...`, `Fix ...`
+
+## Pull requests & Merge policy
```
Comment (GregoryAlbouy) [inline CONTRIBUTING.md:15]: [Suggestion] We could add here some rules we established in the Discord:

- PR Status: draft ➡️ WIP (no reviews); open ➡️ ready for review
- We can request specific reviewers, but anyone can review any PR
- No push on someone else's PR
- PR mergeability: all comments must have been addressed before merge (change in code / agreement in thread / issue create if out of scope)

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Comment (GregoryAlbouy) [inline LICENSE]: Oops 😬

**Comment by c1-ra** [comment]
I am not familiar with setting up CI and Makefiles, so do not hesitate if you have any comment or suggestion!

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+.PHONY: Build
+build:
+	@go build -v main.go
+
+# Docs
+
+.PHONY: docs
+docs:
+	@echo "\033[4mhttp://localhost:9995/pkg/github.com/benchttp/server/\033[0m"
+	@godoc -http=localhost:9995
```
Comment (GregoryAlbouy) [inline Makefile]: [nitpick]
```suggestion
	@godoc -http=localhost:9995

```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- A Pull request must have been tested and reviewed before merging:
+  - all tests must succeed
+  - all comments must have been addressed
+  - **2** reviewers at least must have approved the changes
+
+- PR Status: 
+  - draft   -> WIP (no reviews)
+  - open    -> ready for review
+
+  - No push on someone else's PR
```
Comment (GregoryAlbouy) [inline CONTRIBUTING.md]: [nitpick]
```suggestion
- PR Status: 
  - draft   -> WIP (no reviews)
  - open    -> ready for review

- No push on someone else's PR
```
(so it doesn't end in the "PR Status" list above)

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2021
+Gregory Albouy, Clara Gonnon, Damien Mathieu, Alex Mongeot,
+Thomas Moreira, Kerian Pelat, Icham Sbihi
```
Comment (GregoryAlbouy) [inline LICENSE:3]:
```suggestion
Gregory Albouy, Clara Gonnon, Damien Mathieu, Alex Mongeot,
Thomas Moreira, Kerian Pelat, Marcel Patulacci
```

**Review by GregoryAlbouy — DISMISSED** [review]
🚀

**Review by GregoryAlbouy — APPROVED** [review]

**Review by moreirathomas — APPROVED** [review]

**Pull request merged**